src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java

```diff
@@ -55,18 +61,54 @@ public Uni<List<float[]>> vectorize(
             .register(meterRegistry);
     texts.stream().mapToInt(String::length).forEach(ds::record);
 
+    // Make batches based on provider batch size
+    final List<List<String>> partitions = Lists.partition(texts, batchSize());
+    List<Pair<Integer, List<String>>> partitionedBatches = new ArrayList<>(partitions.size());
+    for (int i = 0; i < partitions.size(); i++) {
+      partitionedBatches.add(Pair.of(i, partitions.get(i)));
+    }
+
     // timer metrics for vectorize call
     Timer.Sample sample = Timer.start(meterRegistry);
     Tags tags = getCustomTags();
-    return embeddingProvider
-        .vectorize(texts, apiKeyOverride, embeddingRequestType)
+    // create a multi
+    return Multi.createFrom()
+        .items(partitionedBatches.stream())
+        .onItem()
+        .transformToUni(
+            batch -> {
+              // call vectorize by the batch id
+              return embeddingProvider.vectorize(
+                  batch.getLeft(), batch.getRight(), apiKeyOverride, embeddingRequestType);
+            })
+        .merge()
+        .collect()
+        .asList()
+        .onItem()
+        .transform(
+            vectorizedBatches -> {
+              // sort it by batch id, this is required because the merge() run the calls in parallel
+              Collections.sort(
+                  vectorizedBatches, (a, b) -> Integer.compare(a.getLeft(), b.getLeft()));
```

Comment:
this is (f.ex) where `getLeft()` is confusing & Record would give more context (we are sorting by batch id)